src/darkslide/generator.py

```diff
@@ -516,40 +556,7 @@ def render(self):
         html = template.render(context)
 
         if self.embed:
-            all_urls = re.findall(r'url\([\"\']?(.*?)[\"\']?\)', html, re.DOTALL | re.UNICODE)
-            img_exts = ('.jpg', '.jpeg', '.png', '.gif', '.svg', '.woff2',
-                        '.woff')
-            img_urls = (url for url in all_urls if url.endswith(img_exts))
-
-            for img_url in img_urls:
-                img_url = img_url.replace('"', '').replace("'", '')
-                if self.theme_dir:
-                    source = os.path.join(self.theme_dir, 'css')
-                else:
-                    source = os.path.join(THEMES_DIR, self.theme, 'css')
```

Comment:
Looks like this is missing in the refactored code?

Reply:
Indeed, but the `else` branch was never executed anyway. 

Now after the refactor, if you look at `__init__()`, the initialization is `self.theme_dir = self.find_theme_dir()`, and `find_theme_dir()` always returns something (or throws a fatal exception).

Besides in `find_theme_dir()` you'll find this:
```python
        elif os.path.exists(os.path.join(THEMES_DIR, theme)):
            theme_dir = os.path.join(THEMES_DIR, theme)
```
So that case is handled.